dashboard/app.py

```diff
@@ -7,24 +7,380 @@
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_name_div(n):
+
+    name = 'Ben Douglas-Griffiths'
+    return f'{name}'
+
+# update name
+
+
+@app.callback(
+    Output(component_id='age-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_age_div(n):
+    age = '24'
+    return f'{age} years old'
+
+# update gender
+
+
+@app.callback(
+    Output(component_id='gender-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_gender_div(n):
+    gender = 'Male'
+    return f'{gender}'
+
+# update weight
+
+
+@app.callback(
+    Output(component_id='weight-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_weight_div(n):
+    weight = '80'
+    return f'{weight}KG'
+
+# update Height
+
+
+@app.callback(
+    Output(component_id='height-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_height_div(n):
+    height = '1.8'
+    return f'{height}m'
+
+
+# Update current users ride's statistics
+
+# update duration
+@app.callback(
+    Output(component_id='duration-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_duration_div(n):
+    duration = '240'
+    return f'{duration} seconds'
+
+# update duration
+@app.callback(
+    Output(component_id='bpm-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_bpm_div(n):
+    bpm = '92'
+    return f'{bpm} BPM'
+
+
+# update total power
+@app.callback(
+    Output(component_id='user-total-power-id', component_property='children'),
+    Input('user-interval-component', 'n_intervals'),
+)
+def update_power_div(n):
+    total_power = '900'
+    return f'{total_power} W'
+
+# User stats updated by the choice of tab
+
+# update user power
+@app.callback(
+    Output(component_id='user-power-id', component_property='children'),
+    [Input('tabs-current-max', 'value'),
+    Input('user-interval-component', 'n_intervals') ]
+)
+def update_bpm_div(tab, n):
+    if tab == 'tab-current':
+        user_power = '30'
+        return f'{user_power} W'
+    user_power = '50'
+    return f'{user_power} W'
+
+# update user RPM
+@app.callback(
+    Output(component_id='rpm-id', component_property='children'),
+    [Input('tabs-current-max', 'value'),
+    Input('user-interval-component', 'n_intervals')]
+)
+def update_rpm_div(tab, n):
+    if tab == 'tab-current':
+        rpm = '30'
+        return f'{rpm} RPM'
+    rpm = '60'
+    return f'{rpm} RPM'
+
+# update user resistance
+@app.callback(
+    Output(component_id='user-resistance-id', component_property='children'),
+    [ Input('tabs-current-max', 'value'),
+     Input('user-interval-component', 'n_intervals') ]
+)
+def update_rpm_div(tab, n):
+    if tab == 'tab-current':
+        resistance = '30'
+        return f'{resistance} Kg'
```

Comment:
Actually, looking further, I'm not sure resistance on bikes *has* units? Might be like tv volume or smth?